zettarepl/transport/base_ssh.py

```diff
@@ -102,10 +102,12 @@ def __init__(self, *args, **kwargs):
 
     def close(self):
         if self._client is not None:
-            self._client.close()
+            threading.Thread(daemon=True, name=f"{threading.current_thread().name}.close_shell",
```

Comment:
Are you creating 2 daemon threads so that you can exit without having to wait on the `close()` methods for these attributes?

Reply:
Yes. Shell `close` method was hanging for one of our users.

Reply:
Are you worried about the fact that `daemon=True` means that the underlying resources for the Shell method could be left in a weird state?

Reply:
@yocalebo unfortunately I am not able to do anything, this is the only `close` method they provide. These underlying resources are discarded by the `Shell` class immediately and not reused anymore.

Reply:
Fair enough